### PR TITLE
Rest framework refactor

### PR DESCRIPTION
**Pull request by RichardZhang01**
Full CRUD for users, receipts, and tags. But not all the features we want are implemented quite yet. Pushing this so we have some updated stuff to work with

### CODE REVIEW

**Comment by RichardZhang01** [comment]
Okay, so this is our rest-framework API sorta more or less working. The glaring issues are: not being able to serve images back to the client, filtering through the api is not implemented, and tags are in a bit of a weird place at the moment, and not tied to any users (which will make listing tags for users to use on the front-end not very intuitive). These are all things that will need some work. In the meantime, you'll find the following endpoints:

/api -- This is the root endpoint for the api. navigating to your localhost at the default port will automatically redirect you to /api regardless. Here you'll find a basic screen with 2 urls to /receipts and /tags. I tried to configure the router to add other urls, but it was a huge pain and I decided not to at the moment. May try to get them in later, but it's not important.

/api/users -- Here you can see a list of all users. Permissions are open so anyone can see access this list without being logged in. 

/api/users/:pk -- this endpoint lists the details for a specific receipt where pk = the user's ID. you can perform update and delete functions at this endpoint/view. Note you can only access this endpoint if you are that user, and get an access denied message if you are not. This means you also have to be logged in.

/api/auth/signup -- this endpoint is post only and you can create a user here. Note this is not a super user. to make a super user you would use the script: python manage.py createsuperuser in your console. This endpoint actually has a view so you'll be able to interact with a form to create a user at this endpoint.

/api/auth/login -- this endpoint is also post only, and you can login using this endpoint. This endpoint, however, does not have a view, so this should only be accessed manually through a post request. Note that if you're using the API directly, there is a login button on the top right corner of all views so you can use that method as well.

/api/auth/logout -- this endpoint is also post only, and like the login endpoint there is no view. You can only post directly to this endpoint. However, if you're logged in, you can also logout from the top right dropdown within the API

/api/receipts -- here you can see a list of all receipts and create a new one. within this view, tags can't be listed in the built-in form which kinda sucks and I tried many ways of including them but they didn't end up working unless I passed the entire tag object in - which wouldn't make sense if the user is meant create new tags when creating/updating a receipt. Anyway, you can bypass this by using JSON directly if you wanted to. (also doing this in the admin panel is a lot simpler and does have selectable tags and the ability to create new ones so maybe go there instead). anyone is able to access the list of receipts even if they're not logged in. However, to create a receipt, you need to be logged in.

/api/receipts/:pk -- similar to users, you can access a receipt here where pk = the receipt's ID. Only the owner of a receipt can update or delete the receipt. I believe if you're authenticated/logged in, you can still see the details however.

/api/users/:pk/receipts -- you can access a list of all receipts that belong to a specific user here where pk = the user's id. I may change the order of the endpoint at a later time. Not sure what makes more sense at the moment. maybe receipts/users/:pk instead?

/api/tags -- List of all tags. if you're logged in, you can also make new ones here. The tag model is very simple at the moment, just a tag_name field.

/api/tags/:pk -- details of a specific tag where pk = the tag's id. Note that only admin users can use this endpoint to delete or update a tag at this time.

/admin -- admin panel. you can login as an admin user here. make sure to create one through the console using python manage.py createsuperuser. This panel is a lot more intuitive than the rest_api, and you can perform filters directly through the tables of data, and perform crud operations a lot easier here. I wish you could convert it directly into an API but anyway, it's there and I've added naming for data entries, columns for specific fields, and filtering.

File: server/core/settings.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = os.getenv('DJANGO_SECRET_KEY')
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = []
+
+
```
Comment (TiciaD) [inline server/core/settings.py]: Do you mind adding this here so I can use GitPod while working on this project? If it's too much of a hassle I can make a separate branch but figured it be easier to throw it in with the rest of the setup
```
CSRF_TRUSTED_ORIGINS = ['https://*.gitpod.io']
CORS_ALLOWED_ORIGIN_REGEXES = [
    r"^https://.*\.gitpod\.io$",
]
```
It doesn't need to be anywhere specific I just randomly chose under the ALLOWED_HOSTS line

**Review by TiciaD — APPROVED** [review]
Looks great Richard! Thanks for getting so much of the backend up and working, this is a great start to build off of! I only had one comment but it's not anything wrong with what you've added just some extra setup for my dev environment

File: server/core/settings.py (hunk 1 above)
Comment (RichardZhang01) [reply server/core/settings.py]: Added. Thanks

**Pull request merged**